### PR TITLE
[Enhancement] Migrates to View Pager 2

### PR DESCRIPTION
**Pull request by guchito9**
Migrates payment method pager in one tap to ViewPager2

## Motivación y Contexto
ViewPager1 has been deprecated for a long time now and we have a crash related to it that we hope to fix with this.

## Descripción
- Migrates to ViewPager2 in OneTap.
- Adapts scroll indicator to work with ViewPager2.
- Uses page transform for the negative margin removing the previous hack to have it.
- Makes a small refactor to how we set LOW_RES mode in installmentsV1.

## Tipo de cambio (para el release manager)
- [ ] Breaking change (Fix o feature que cambia una funcionalidad existente o rompe firmas)

- [ ] Mi cambio afecta a los integradores internos

### CODE REVIEW

**Review by daniivillegas — DISMISSED** [review]

**Review by francofusaro — DISMISSED** [review]
Left two minor comments 😄

**Review by daniivillegas — APPROVED** [review]

File: px-checkout/src/main/java/com/mercadopago/android/px/internal/features/one_tap/slider/pager/PaymentMethodPagerConfigurator.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import android.graphics.Rect
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+import androidx.viewpager2.widget.ViewPager2
+import com.mercadopago.android.px.internal.features.one_tap.slider.PaymentMethodPageTransformer
+
+internal object PaymentMethodPagerConfigurator {
+
+    fun configure(pager: ViewPager2, itemPadding: Int) {
```
Comment (jorGonzalez291292) [inline px-checkout/src/main/java/com/mercadopago/android/px/internal/features/one_tap/slider/pager/PaymentMethodPagerConfigurator.kt:11]: For some reason I feel like this logic may be part of our viewPager2.

File: px-checkout/src/main/java/com/mercadopago/android/px/internal/features/one_tap/slider/pager/ViewPagerAttacher.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                indicator.reattach()
+            }
+        }
+        attachedAdapter.registerAdapterDataObserver(adapterDataObserver)
+        onPageChangeCallback = object : OnPageChangeCallback() {
+            private var idleState = true
+            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
+                super.onPageScrolled(position, positionOffset, positionOffsetPixels)
+                // ViewPager may emit negative positionOffset for very fast scrolling
+                val offset: Float = when {
```
[the remark this answers is not in the document]
Comment (guchito9) [reply px-checkout/src/main/java/com/mercadopago/android/px/internal/features/one_tap/slider/pager/ViewPagerAttacher.kt:34]: I swear I didn't do nothing in this code besides migrating to Kotlin

File: px-checkout/src/main/java/com/mercadopago/android/px/internal/features/one_tap/slider/pager/PaymentMethodPagerConfigurator.kt (hunk 1 above)
Comment (guchito9) [reply px-checkout/src/main/java/com/mercadopago/android/px/internal/features/one_tap/slider/pager/PaymentMethodPagerConfigurator.kt:11]: ViewPager2 class is final

File: px-checkout/src/main/java/com/mercadopago/android/px/internal/features/one_tap/slider/pager/ViewPagerAttacher.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (guchito9) [reply px-checkout/src/main/java/com/mercadopago/android/px/internal/features/one_tap/slider/pager/ViewPagerAttacher.kt:34]: Yes, the original code wasn't the greatest thing ever, would do the change you propose

**Review by francofusaro — DISMISSED** [review]

**Review by daniivillegas — DISMISSED** [review]

**Comment by jorGonzalez291292** [comment]
@guchito9 Arreglate el changelog y lo apruebo.

**Pull request merged**